datacube_stats/main.py

```diff
@@ -189,97 +186,101 @@ def _log_setup():
         #: and will define spatial and temporal boundaries, as well as statistical operations to be run.
-        self.task_generator = None
+        self.task_generator = select_task_generator(config['input_region'],
+                                                    self.storage, self.filter_product)
 
         #: A class which knows how to create and write out data to a permanent storage format.
         #: Implements :class:`.output_drivers.OutputDriver`.
-        self.output_driver = None
+        self.output_driver = _prepare_output_driver(self.storage)
 
         #: An open database connection
-        self.index = None
-
-        self.global_attributes = None
-        self.var_attributes = None
+        self.index = index
 
-    @classmethod
-    def from_configuration_file(cls, config, index=None, tile_indexes=None, output_location=None, year=None):
-        """
-        Create a StatsApp to run a processing job, based on a configuration file
-
-        :param config: dictionary based configuration
-        :param index: open database connection
-        :param tile_indexes: list of tiles for a gridded job. (useful for debugging)
-        :return: read to run StatsApp
-        """
-        input_region = config.get('input_region')
-
-        if tile_indexes is not None:
-            if input_region is None:
-                input_region = {'tiles': tile_indexes}
-            elif 'geometry' in input_region:
-                input_region.update({'tiles': tile_indexes})
-            elif 'from_file' not in input_region:
-                input_region = {'tiles': tile_indexes}
-
-        if year is not None:
-            if 'date_ranges' not in config:
-                config['date_ranges'] = {}
-
-            config['date_ranges']['start_date'] = '{}-01-01'.format(year)
-            config['date_ranges']['end_date'] = '{}-01-01'.format(year + 1)
-
-        stats_app = cls()
-        stats_app.index = index
-        stats_app.config_file = config
-        stats_app.storage = config['storage']
-        stats_app.sources = config['sources']
-        stats_app.output_product_specs = config['output_products']
-        # Write files to current directory if not set in config or command line
-        stats_app.location = output_location or config.get('location', '')
-        stats_app.computation = config.get('computation', {})
-        stats_app.date_ranges = _configure_date_ranges(index, config)
-        stats_app.filter_product = config.get('filter_product', {})
-        stats_app.task_generator = select_task_generator(input_region, stats_app.storage, stats_app.filter_product)
-        stats_app.output_driver = _prepare_output_driver(stats_app.storage)
-        stats_app.global_attributes = config.get('global_attributes', {})
-        stats_app.var_attributes = config.get('var_attributes', {})
-
-        return stats_app
+        self.global_attributes = config['global_attributes']
+        self.var_attributes = config['var_attributes']
```

Comment:
This is nice, if a `StatsApp` object has been created, it can be assumed ready to go.

Do you think it would make sense to call `.validate()` at the end of `.__init__()`, or does it still make sense to separate the steps?